File: source/clibCore/Interfaces/mainCsvCompatibilityClass.vb

Option Explicit On
Option Strict On

Imports System.Runtime.InteropServices
'
Namespace Contensive.Core
    ''' <summary>
    ''' This class provides a compatibility api for legacy active-script addons that use the "cclib" object for main class nad "csv" for the csv object
    ''' </summary>
    <ComVisible(True)>
    <ComClass(mainCsvCompatibilityClass.ClassId, mainCsvCompatibilityClass.InterfaceId, mainCsvCompatibilityClass.EventsId)>
    Public Class mainCsvCompatibilityClass
        '
#Region "COM GUIDs"
        Public Const ClassId As String = "D9099AAE-3FCB-4398-B94C-19EE7FA97B2B"
        Public Const InterfaceId As String = "CE342EA5-339F-4C31-9F90-F878F527E17A"
        Public Const EventsId As String = "21D9D0FB-9B5B-43C2-A7A5-3C84ABFAF90A"
#End Region
        '
        Private cpCore As coreClass

        Public Sub New(cpCore As coreClass)
            Me.cpCore = cpCore
        End Sub

        '
        '====================================================================================================
        '
        Public Function EncodeContent9(Source As Object, i0 As Integer, s2 As String, i3 As Integer, i4 As Integer, b5 As Boolean, b6 As Boolean, b7 As Boolean, b8 As Boolean, b9 As Boolean, b10 As Boolean, s11 As String, s12 As String, b13 As Boolean, i14 As Integer, s15 As String, i16 As Integer) As String
            Return cpCore.html.encodeContent10(Controllers.genericController.encodeText(Source), 0, "", 0, 0, False, False, False, True, True, False, "", "", False, 0, "", Contensive.BaseClasses.CPUtilsBaseClass.addonContext.ContextSimple, False, Nothing, False)
        End Function
    End Class
End Namespace

File: source/clibCoreCs/Interfaces/mainCsvCompatibilityClass.cs

using System;
using System.Reflection;
using System.Xml;
using System.Diagnostics;
using System.Linq;
using System.Text.RegularExpressions;
using System.Collections.Generic;
using System.Data;
using System.Data.SqlClient;
using Contensive.Core;
using Contensive.Core.Models.Entity;
using Contensive.Core.Controllers;
using static Contensive.Core.Controllers.genericController;
using static Contensive.Core.constants;
// 
namespace Contensive.Core {
    public class mainCsvScriptCompatibilityClass {
        public const string ClassId = "D9099AAE-3FCB-4398-B94C-19EE7FA97B2B";
        public const string InterfaceId = "CE342EA5-339F-4C31-9F90-F878F527E17A";
        public const string EventsId = "21D9D0FB-9B5B-43C2-A7A5-3C84ABFAF90A";
        // 
        private coreClass cpCore;
        public mainCsvScriptCompatibilityClass(coreClass cpCore) {
            this.cpCore = cpCore;
        }
        
        // 
        // ====================================================================================================
        // 
        public string EncodeContent9(
                    object Source, 
                    int i0, 
                    string s2, 
                    int i3, 
                    int i4, 
                    bool b5, 
                    bool b6, 
                    bool b7, 
                    bool b8, 
                    bool b9, 
                    bool b10, 
                    string s11, 
                    string s12, 
                    bool b13, 
                    int i14, 
                    string s15, 
                    int i16) {
            return cpCore.html.convertActiveContentToHtmlForWysiwygEditor(Controllers.genericController.encodeText(Source));
        }
    }
}